### PR TITLE
Support arm64v8 

### PR DESCRIPTION
**Pull request by LukasBrand**
https://hivemq.kanbanize.com/ctrl_board/22/cards/23983/details/

Addresses: #434

### CODE REVIEW

File: src/integrationTest/java/com/hivemq/cli/commands/cli/TestBrokerCommandDefaultIT.java [hunk 1]
```diff
@@ -18,38 +18,29 @@
...
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4"));
-
-    @BeforeAll
-    static void beforeAll() {
-        hivemq.start();
-    }
+    @Container
+    private final @NotNull HiveMQContainer hivemq = new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"));
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/commands/cli/TestBrokerCommandDefaultIT.java:37]: Before, the hivemq container was used for all tests in this class (started beforeAll, stopped afterAll).
Now, every test gets a new hivemq container.
This is the case for multiple tests (I will not duplicate this comment to all occurrences).

File: build.gradle.kts [hunk 2]
```diff
@@ -595,17 +605,49 @@ gitPublish {
...
+                    contents {
+                        from(configurations.runtimeClasspath)
+                        into("app/libs")
+                    }
+                }
+                layer("jar") {
+                    contents {
+                        from(tasks.jar)
+                        into("app/classpath")
+                        rename(".*", "${project.name}-${project.version}.jar")
```
Comment (SgtSilvio) [inline build.gradle.kts:639]:
```suggestion
```

File: build.gradle.kts [hunk 3]
```diff
@@ -595,17 +605,49 @@ gitPublish {
...
+            parentImages {
+                add("library:eclipse-temurin:sha256!603d23272e30bbefa9e7c436a7165c6303b9c67e27aae07472d8ddc748fe96a2") // 21.0.2_13-jre-jammy
+            }
+            config {
+                entryPoint.add("java")
+                entryPoint.addAll(application.applicationDefaultJvmArgs)
+                entryPoint.addAll(
+                    "-cp",
+                    "/app/classpath/*:/app/libs/*",
+                )
```
Comment (SgtSilvio) [inline build.gradle.kts:625]:
```suggestion
                entryPoint.addAll("-cp", "/app/classpath/*:/app/libs/*")
```

File: build.gradle.kts [hunk 4]
```diff
@@ -595,17 +605,49 @@ gitPublish {
...
+                layer("jar") {
+                    contents {
+                        from(tasks.jar)
+                        into("app/classpath")
+                        rename(".*", "${project.name}-${project.version}.jar")
+                    }
+                }
+                layer("resources") {
+                    contents {
+                        from("src/distribution")
```
Comment (SgtSilvio) [inline build.gradle.kts:644]:
```suggestion
                        from("src/distribution")
                        into("app")
```

File: build.gradle.kts (hunk 2 above)
Comment (SgtSilvio) [reply build.gradle.kts:639]: nvm

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterDefaultIT.java [hunk 5]
```diff
@@ -44,28 +44,20 @@
...
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@Testcontainers
 class Mqtt3FeatureTesterDefaultIT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq");
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterDefaultIT.java:52]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"));
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterQos0IT.java [hunk 6]
```diff
@@ -18,40 +18,32 @@
...
 
+@Testcontainers
 class Mqtt3FeatureTesterQos0IT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/qos0-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterQos0IT.java:37]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterQos1IT.java [hunk 7]
```diff
@@ -18,40 +18,32 @@
...
 
+@Testcontainers
 class Mqtt3FeatureTesterQos1IT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/qos1-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterQos1IT.java:37]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterRestrictedIT.java [hunk 8]
```diff
@@ -39,29 +39,21 @@
...
 
+@Testcontainers
 class Mqtt3FeatureTesterRestrictedIT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/restricted-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt3FeatureTesterRestrictedIT.java:47]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterDefaultIT.java [hunk 9]
```diff
@@ -29,40 +29,32 @@
...
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@Testcontainers
 class Mqtt5FeatureTesterDefaultIT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq");
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterDefaultIT.java:51]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"));
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterQos0IT.java [hunk 10]
```diff
@@ -18,40 +18,32 @@
...
 
+@Testcontainers
 class Mqtt5FeatureTesterQos0IT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/qos0-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterQos0IT.java:37]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterQos1IT.java [hunk 11]
```diff
@@ -18,40 +18,32 @@
...
 
+@Testcontainers
 class Mqtt5FeatureTesterQos1IT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/qos1-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterQos1IT.java:37]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterRestrictedIT.java [hunk 12]
```diff
@@ -39,29 +39,21 @@
...
 
+@Testcontainers
 class Mqtt5FeatureTesterRestrictedIT {
 
-    private static final @NotNull HiveMQTestContainerExtension hivemq =
-            new HiveMQTestContainerExtension(DockerImageName.parse("hivemq/hivemq4")).withHiveMQConfig(MountableFile.forClasspathResource(
-                    "mqtt/test/restricted-config.xml"));
+    @Container
+    private final @NotNull HiveMQContainer hivemq =
+            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4")).withNetworkAliases("hivemq")
```
Comment (SgtSilvio) [inline src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterRestrictedIT.java:47]:
```suggestion
            new HiveMQContainer(OciImages.getImageName("hivemq/hivemq4"))
```

File: src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.hivemq.cli.utils.broker.assertions.ConnectAssertion.assertConnectPacket;
+import static com.hivemq.cli.utils.broker.assertions.PublishAssertion.assertPublishPacket;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Testcontainers
```
Comment (SgtSilvio) [inline src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java:27]:
```suggestion
```

File: src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    @AfterEach
+    void tearDown() {
+        mqttCli.stop();
+    }
+
+    @ParameterizedTest
+    @Timeout(value = 3, unit = TimeUnit.MINUTES)
+    @ValueSource(chars = {'3', '5'})
+    void test_successfulConnectAndPublish(final char mqttVersion) throws Exception {
+        final List<String> publishCommand1 = List.of("pub",
```
Comment (SgtSilvio) [inline src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java:45]:
```suggestion
        final String[] publishCommand1 = { "pub",
```

File: src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+                String.valueOf(hivemq.getMqttPort()),
+                "-V",
+                String.valueOf(mqttVersion),
+                "-i",
+                "cliTest",
+                "-t",
+                "test",
+                "-m",
+                "message",
+                "-d");
```
Comment (SgtSilvio) [inline src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java:58]:
```suggestion
                "-d" };
```

File: src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+        mqttCli.withCommand(publishCommand1.toArray(new String[0])).start();
+
+        final ExecutionResult executionResult = MqttCli.execute(publishCommand1);
```
Comment (SgtSilvio) [inline src/systemTest/java/com/hivemq/cli/commands/distribution/DockerImageST.java:61]: withCommand + execute?

File: src/integrationTest/java/com/hivemq/cli/commands/cli/TestBrokerCommandDefaultIT.java (hunk 1 above)
Comment (LukasBrand) [reply src/integrationTest/java/com/hivemq/cli/commands/cli/TestBrokerCommandDefaultIT.java:37]: This was changed on purpose to remove static objects.

**Review by SgtSilvio — APPROVED** [review]

File: src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterDefaultIT.java [hunk 17]
```diff
@@ -29,40 +29,31 @@
...
-
     @BeforeEach
     void setUp() {
-        mqtt5FeatureTester = new Mqtt5FeatureTester(hivemq.getHost(), hivemq.getMqttPort(), null, null, null, 3);
-    }
-
-    @AfterAll
-    static void afterAll() {
-        hivemq.stop();
+        mqtt5FeatureTester = new Mqtt5FeatureTester(hivemq.getHost(), hivemq.getMqttPort(), null, null, null, 10);
```
[the remark this answers is not in the document]
Comment (LukasBrand) [reply src/integrationTest/java/com/hivemq/cli/mqtt/test/Mqtt5FeatureTesterDefaultIT.java:56]: The timeout of 3 was too small so it sometimes created flaky behavior.

**Pull request merged**